common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -273,6 +286,7 @@ public long getPrefix() {
    * @param until the position after last code point, exclusive.
    */
   public UTF8String substring(final int start, final int until) {
+    int numBytes = numBytes();
```

Comment:
see https://github.com/apache/spark/pull/19222/files#r171586767

Sorry for the back and forth, but just adding a duplicated `numBytes` field seems to make our life easier.

Reply:
Got it. I will leave this decision in the comment.